rescript/plugin_setup.py

```diff
@@ -483,20 +485,44 @@
                            'the rank-lineage string in the \'id\' '
                            'column is replaced by the rank-lineage '
-                           'string of the provided column name. All '
+                           'string in a second named column. All '
                            'items in the \'id\' column must be '
                            'unique!',
-        'regex': 'Toggle regular expressions. If \'False\' only full '
-                 'string matching is performed.'},
+        'search_strings': 'Only used in conjuntion with '
+                          '\'replacement-strings\'. Each string in this list '
+                          'is searched for and replaced with a string in the '
+                          'list of \'replace-ment-strings\'. '
+                          'That is the first string in \'search-strings\' is '
+                          ' replaced with the first string in '
+                          '\'replacement-strings\', and so on. The number of '
+                          '\'search-strings\' must be equal to the number of '
+                          'replacement strings.',
+        'replacement_strings': 'Only used in conjuntion with '
+                               '\'search-strings\'. This must contain the '
+                               'same number of replacement strings as search '
+                               'strings. See \'search-strings\' parameter '
+                               'text for more details.',
+        'use_regex': 'Toggle regular expressions. By default, only litereal '
+                     'substring matching is performed.'},
     output_descriptions={
         'edited_taxonomy': 'Taxonomy in which the original rank-lineage '
                            'strings were replaced by user-supplied '
                            'rank-lineage strings.'
     },
     name='Replaces existing taxonomy strings with those provided',
-    description=('This method allows the user to edit taxonomy strings. '
-                 'This is often used to fix inconsistently applied '
-                 'nomenclature for identical seqeunces.')
+    description=('A method that allows the user to edit taxonomy strings. '
+                 'This is often used to fix inconsistent and/or '
+                 'inccorect taxonomic annotations. The user can either '
+                 'provide two separate lists of strings, i.e. '
+                 '\'search-strings\', and \'replacement-strings\', '
+                 'on the command line, or a single tab-delimited '
+                 'replacement map file containing a list of these strings. '
+                 'In both cases the number of search strings must match the '
+                 'number of replacement strings. That is the first string '
+                 'in \'search-strings\' is replaced with the first string '
+                 'in \'replacement-strings\', and so on. Finally,if both '
```

Comment:
does one or the other need to take precedence? maybe you could just merge the replacement dictionaries to allow use of both?

Reply:
I think we'll need to workout why passing in regex strings at the command line require `\\` escape characters, while the regex patterns in the mapping file use `\` to escape, before we merge these.

Reply:
It will now merge the dictionaries if the user supplies the stings via the mapping file and the command line.